db/db_impl/db_impl_open.cc

```diff
@@ -516,6 +515,11 @@ Status DBImpl::Recover(
   } else {
     s = SetIdentityFile(env_, dbname_, db_id_);
   }
+  if (s.ok() && !read_only) {
+    // FinishRecovery may switch to new MANIFEST, which requires db_id to be
+    // set.
+    s = FinishRecovery();
```

Comment:
I wonder whether the writing of db_id can be included in `FinishRecovery()`.

Reply:
`FinishRecovery` feels too general, I renamed it to `DeleteUnreferencedSstFilesAndSwitchToNewManifest`, although a long name, but the intention is more clear. 

Also moved the db id related logic to a new method `SetDBId`.